pkgs/build-support/writers/scripts.nix

```
@@ -555,53 +556,25 @@ rec {
     :::
   */
   writeBabashka =
-    name: argsOrScript:
-    if lib.isAttrs argsOrScript && !lib.isDerivation argsOrScript then
-      makeScriptWriter (
-        argsOrScript
-        // {
-          interpreter = "${lib.getExe pkgs.babashka}";
-          check = "${lib.getExe pkgs.clj-kondo} --lint";
-        }
-      ) name
-    else
-      makeScriptWriter {
-        interpreter = "${lib.getExe pkgs.babashka}";
-        check = "${lib.getExe pkgs.clj-kondo} --lint";
-      } name argsOrScript;
+    name:
+    {
+      makeWrapperArgs ? [ ],
+      babashka ? pkgs.babashka-unwrapped,
+      check ? "${lib.getExe pkgs.clj-kondo} --lint",
+      ...
+    }@args:
+    makeScriptWriter (
+      (builtins.removeAttrs args [
+        "babashka"
+      ])
+      // {
+        interpreter = "${lib.getExe babashka}";
+      }
+    ) name;
```

Why not something like this to avoid the empty attribute set?

```suggestion
    name: argsOrScript:
    let babashka = argsOrScript.babashka or pkgs.babashka-unwrapped;
        defaultCheck = "${lib.getExe pkgs.clj-kondo} --lint";
    in
    if lib.isAttrs argsOrScript && !lib.isDerivation argsOrScript then
      makeScriptWriter (
        {
          interpreter = "${lib.getExe babashka}";
          check = defaultCheck;
        } // (builtins.removeAttrs argsOrScript ["babashka"])
      ) name
    else
      makeScriptWriter {
        interpreter = "${lib.getExe babashka}";
        check = defaultCheck;
      } name argsOrScript;

```